.golangci.yml

```diff
@@ -0,0 +1,41 @@
+run:
+  # timeout for analysis, e.g. 30s, 5m, default is 1m
+  timeout: 5m
+
+  # Skip linting _test.go files
+  tests: false
+
+  # Enables skipping of directories:
+  # vendor$, third_party$, testdata$, examples$, Godeps$, builtin$
+  skip-dirs-use-default: true
+
+  # Allow multiple parallel golangci-lint instances running.
+  # If false (default) - golangci-lint acquires file lock on start.
+  allow-parallel-runners: true
+
+linters:
+  enable:
+    - bodyclose
+    - deadcode
+    - errcheck
+    - gocyclo
+    - gofmt
+    - gosimple
+    - govet
+    - ineffassign
+    - lll
+    - staticcheck
+    - structcheck
+    - typecheck
+    - unused
+    - varcheck
+
+linters-settings:
+  gocyclo:
+    min-complexity: 20 # default is 30.
+
+  lll:
+    # max line length, lines longer will be reported. Default is 120.
+    line-length: 100
+    # tab width in spaces. Default to 1.
+    tab-width: 4
```

Comment:
:sob:

Reply:
:ballot_box:

Reply:
just since I've come across the related line-length setting... these settings here are directly against the motivating factors of the language > https://github.com/golang/go/issues/40028 < fmt and IMO this should be reversed...